adaptive_ai/adaptive_ai/services/domain_classifier.py

```
@@ -0,0 +1,361 @@
+                domain_probabilities=domain_probabilities,
+            )
+            results.append(result)
+
+        return results
+
+
+class DomainClassifier:
+    def __init__(
+        self, lit_logger: Any = None, cache_size: int = 1000, cache_ttl: int = 3600
+    ) -> None:
+        self.lit_logger: Any = lit_logger
+        self.cache_size: int = cache_size
+        self.cache_ttl: int = cache_ttl
+        self.cache: OrderedDict[str, tuple[DomainClassificationResult, float]] = (
+            OrderedDict()
+        )  # LRU cache for domain classification results
+        self.cache_hits: int = 0
+        self.cache_misses: int = 0
+
+        # Load NVIDIA domain classifier following official documentation
+        try:
+            self.config = AutoConfig.from_pretrained("nvidia/domain-classifier")
+            self.tokenizer = AutoTokenizer.from_pretrained("nvidia/domain-classifier")
+
+            # Load the model using the CustomModel class from official docs with safetensors
+            self.model = CustomModel.from_pretrained(
+                "nvidia/domain-classifier", use_safetensors=True
+            )
+            self.model.eval()
+
+            self.log(
+                "domain_classifier_weights_loaded",
+                {"status": "success", "source": "nvidia/domain-classifier"},
+            )
+        except Exception as e:
+            self.log("domain_classifier_weights_load_failed", {"error": str(e)})
+            # Fallback to custom implementation if NVIDIA loading fails
+            self.log(
+                "domain_classifier_fallback", {"status": "using_custom_implementation"}
+            )
+            # Use the original custom implementation as fallback
+            self._setup_custom_fallback(lit_logger)
+
+    def log(self, key: str, value: Any) -> None:
+        if self.lit_logger:
+            self.lit_logger.log(key, value)
+
+    def _setup_custom_fallback(self, lit_logger: Any) -> None:
```

delete this custom domain classifier no need for fallbacks, this again makes errors silently pass through